app/models/knight.rb

```diff
@@ -22,4 +22,16 @@ def legal_move?(x, y)
     end
   end
 
+  def obstructed_path?(row, col)
+  end
+
+  def valid_move?(row, col)
+  end
+
+  def piece_where_moving_to?(row, col)
+  end
+
+  def piece_still_on_board?(row, col)
+  end
+
```

Comment:
These 3 methods should be identical for all pieces so can be moved up to the superclass.